tests/test_external_connection.py

```diff
@@ -0,0 +1,120 @@
+                "base-area/register_participant/response"
+            assert self.external_connection.redis_db.publish.call_args_list[i][0][1] == \
+                json.dumps({"available_publish_channels": [
+                   f"base-area/{slugify(created_area)}/offer",
+                   f"base-area/{slugify(created_area)}/offer_delete",
+                   f"base-area/{slugify(created_area)}/offer_accept"
+                ], "available_subscribe_channels": [
+                   f"base-area/{slugify(created_area)}/offers",
+                   f"base-area/{slugify(created_area)}/offer/response",
+                   f"base-area/{slugify(created_area)}/offer_delete/response",
+                   f"base-area/{slugify(created_area)}/offer_accept/response"
+                ]})
+
+    def test_unregister_areas_deletes_children_from_area(self):
+        area_list = ["kreuzberg", "friedrichschain", "prenzlauer berg"]
+        self.external_connection.areas_to_register = area_list
+        self.external_connection.register_new_areas()
+        assert set(ch.name for ch in self.area.children) == set(area_list)
+        # Keep track how many times the publish method was called when setting up the areas
+        call_count = self.external_connection.redis_db.publish.call_count
+
+        self.external_connection.areas_to_unregister = ["prenzlauer berg"]
+        self.external_connection.unregister_pending_areas()
+        assert len(self.area.children) == 2
+        assert self.external_connection.redis_db.publish.call_count == call_count + 1
+        self.external_connection.redis_db.publish.assert_called_with(
+            "base-area/unregister_participant/response", json.dumps({"response": "success"}))
+
+        self.external_connection.areas_to_unregister = ["friedrichschain"]
+        self.external_connection.unregister_pending_areas()
+        assert len(self.area.children) == 1
+        assert self.external_connection.redis_db.publish.call_count == call_count + 2
+        self.external_connection.redis_db.publish.assert_called_with(
+            "base-area/unregister_participant/response", json.dumps({"response": "success"}))
+
+    def test_unregister_area_that_does_not_exist_returns_an_error(self):
+        area_list = ["kreuzberg", "friedrichschain", "prenzlauer berg"]
+        self.external_connection.areas_to_register = area_list
+        self.external_connection.register_new_areas()
+        assert set(ch.name for ch in self.area.children) == set(area_list)
+
+        self.external_connection.areas_to_unregister = ["schoneberg"]
+        self.external_connection.unregister_pending_areas()
+        assert len(self.area.children) == 3
+        self.external_connection.redis_db.publish.assert_called_with(
+            "base-area/unregister_participant/response", json.dumps({"response": "failed"}))
+
+    def test_subscribe_channel_list_fetches_correct_channel_names(self):
```

Comment:
Dont you already test this implicitly in  test_register_new_areas_creates_new_child_areas_from_buffer ?

Reply:
True, I wanted to have a specific test though that validates that the method that fetches the channels works as expected. If you think this is an overkill I can remove this test.